### PR TITLE
Add support to loop a test a number of times

### PR DESCRIPTION
**Pull request by calvaris**
This can be helpful when you are trying to debug a test or bisect something against this tests.

### CODE REVIEW

File: instructions.html [hunk 1]
```diff
@@ -58,7 +58,9 @@
         <span class='code'>
           <a href="tip.html?loop=true">
             tip.html?loop=true</a>
-        </span> to turn on loop.
+        </span> to turn on loop or <a href="tip.html?loop=10">
```
Comment (tdedecko) [inline instructions.html]: The anchor tag should be contained in a <span class='code'> tag

File: js/harness/test.js [hunk 2]
```diff
@@ -114,6 +114,7 @@ var TestRunner = function(testSuite, testsMask, testSuiteVer) {
   this.timeouts = createTimeoutManager(createLogger(this.log.bind(this)));
   this.lastResult = 'pass';
   this.testSuiteVer = testSuiteVer;
+  this.runs = 0;
```
Comment (tdedecko) [inline js/harness/test.js]: rename runCount

File: js/harness/test.js [hunk 3]
```diff
@@ -274,7 +275,7 @@ TestRunner.prototype.onfinished = function() {
   }
 
   var keepRunning = (!stoponfailure || this.lastResult === 'pass') &&
-      loop && (this.testView.anySelected() || this.numOfTestToRun === 1);
+      (++this.runs < loop) && (this.testView.anySelected() || this.numOfTestToRun === 1);
```
Comment (tdedecko) [inline js/harness/test.js]: Please increment runCount outside of this complex expression.

**Comment by tdedecko** [comment]
This change looks pretty good. I would like to pull this in. I just have a few small suggestions. Thanks for contributing.

File: js/harness/main.js [hunk 4]
```diff
@@ -54,7 +54,15 @@ var configureHarness = function(config) {
   window.currentTestType = config.testType;
   TestBase.timeout = config.timeout;
   window.logging = config.disableLog !== 'true';
-  window.loop = config.loop === 'true';
+  if (config.loop === 'true')
+    window.loop = Number.POSITIVE_INFINITY
+  else {
+    window.loop = Number.parseInt(config.loop);
```
Comment (tdedecko) [inline js/harness/main.js]: assign the parsed value to a temporary var instead of window.loop

**Comment by calvaris** [comment]
Thanks @tdedecko, I already honored your comments.

Taking advantage of having your attention I wanted to tell you that I found that the encrypted video examples for ClearKey for non-webm are encrypted with PlayReady instead of ClearKey and that is causing big issues as decryption is not possible. Sorry to comment it in this unrelated PR but you don't have the issues enabled.

**Comment by calvaris** [comment]
To be more precise about the encryption. They seem to be done with PlayReady and Widevine, but not ClearKey

**Comment by tdedecko** [comment]
The changes overall look good. I just want to test this out. I'll be able to do next week.

As for, encryption issue... You need to provide more information to back that up. Please explain what you think the issue is in terms of the moov atom and pssh headers. It would probably be best to reach out to our team to help you diagnose those issues.

**Comment by calvaris** [comment]
Thanks for having a look at the patches. Take your time, they are cosmetic :)

About the encrypted files

VideoStreamYTCenc: ['media/oops_cenc-20121114-145-no-clear-start.mp4', yields system ids
edef8ba9-79d6-4ace-a3c8-27dcd51d21ed Widevine in pssh box
9a04f079-9840-4286-ab92-e65be0885f95 PlayReady in pssh box
58147ec8-0423-4659-92e6-f52c5ce8c3cc ClearKey in pssh box

Of course this works because the file is encrypted with ClearKey, we have the key and we can play. This works for test 1 and test 4.

In the case of test 3 that uses VideoNormalClearKey: ['media/car_cenc-20120827-86.mp4' we are getting:
9a04f079-9840-4286-ab92-e65be0885f95 PlayReady in pssh box
edef8ba9-79d6-4ace-a3c8-27dcd51d21ed Widevine in pssh box
9a04f079-9840-4286-ab92-e65be0885f95 PlayReady in piff box
but no uuid of ClearKey so file cannot be decrypted.

Same happens for test 2 with AudioNormalClearKey: ['media/car_cenc-20120827-8c.mp4'

So there are some mp4 files at streamDef.js that are defined there and are wrongly encrypted which makes them impossible to decrypt.

I think what would be interesting is that you people have a look at streamDef.js and check how the files are encrypted there.

Thank you very much for your help!

**Comment by calvaris** [comment]
Hi, @tdedecko, any change we can get this merged?

Btw, did you have time to check if the files for tests 2 and 3 are correctly encrypted with ClearKey, cause as I stated in my previous comment, I believe they are not and this prevents people from passing the tests.

File: js/harness/main.js [hunk 5]
```diff
@@ -54,7 +54,17 @@ var configureHarness = function(config) {
   window.currentTestType = config.testType;
   TestBase.timeout = config.timeout;
   window.logging = config.disableLog !== 'true';
-  window.loop = config.loop === 'true';
+  var loop;
+  if (config.loop === 'true')
+    loop = Number.POSITIVE_INFINITY
```
Comment (tdedecko) [inline js/harness/main.js]: Add a semicolon to the end of this line

File: js/harness/test.js [hunk 6]
```diff
@@ -273,8 +274,9 @@ TestRunner.prototype.onfinished = function() {
              this.longestTimeRatio + ' of its timeout.');
   }
 
+  ++this.runCount;
   var keepRunning = (!stoponfailure || this.lastResult === 'pass') &&
-      loop && (this.testView.anySelected() || this.numOfTestToRun === 1);
+      (this.runCount < loop) && (this.testView.anySelected() || this.numOfTestToRun === 1);
```
Comment (tdedecko) [inline js/harness/test.js]: Make this line less than 80 characters.

**Comment by tdedecko** [comment]
Please resolve the two small comments and I'll merge this in.

I also added bug reporting to the repository. Please report bugs there.

File: js/harness/test.js (hunk 6 above)
Comment (calvaris) [inline js/harness/test.js]: Done, for the same price I fixed the indentation too ;)

**Pull request merged**